core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -330,19 +332,21 @@ object SparkSubmit extends CommandLineUtils {
     args.archives = Option(args.archives).map(resolveGlobPaths(_, hadoopConf)).orNull
 
     // In client mode, download remote files.
+    var localPrimaryResource: String = null
+    var localJars: String = null
+    var localPyFiles: String = null
```

Comment:
hmm, so I guess the original one may have been to add things to the classpath or python path or the actual python file to run, but to me it seems inconsistent that we download some of these things and not others.  

For example, my primary resource program that gets automatically downloaded from hdfs requires file X for the driver to read, but we don't download that automatically.  So user has to manually download X (and possibly archive Y) but doesn't have to do the jars or primary resource.  Normally if the user used --files they expect that file to be in ./filename.   Of course I guess the way this is downloading it wouldn't be in ./ either.  That is the same case for archives as well.

It seems to be me this was missed in the design of this feature.  Even in their example using docker and marathon to run spark-submit in client mode won't work if they specify --files now and their driver needs that file.  to me this is the same as needing the jar.  They would have to package it in the docker image, thus defeating the purpose of the feature.  I'm sure less people use --files in this case, but we shouldn't have inconsistent behavior.

if the user can't rely on spark to download everything they need to figure out which parts to download themselves or just do as is now and setup everything on the gateway in which case the downloads happening here are just extra.  

This to me just sounds like I need to explain things to users even more and they are going to be confused or I just tell them to run like they are now.  

anyway, I kind of think this feature shouldn't have went in if we can't make it act just like it was running on the cluster. But I think making it run like on the cluster is really hard with out changing users running directory, which again could break other things.  If we download things to ./ then you could do bad things as well that the user doesn't expect.

Honestly now I'm leaning towards we should have this configurable with default off. I see the one use case it was added for, but even that is incomplete in my opinion.

Adding the original author and reviewers: @loneknightpy @cloud-fan @gatorsmile 

thoughts?